worker.go

```diff
@@ -125,42 +128,44 @@ func (w *Worker) Run(ctx context.Context) {
 		setSkipTLS(subscriberOptions)
 	}
 
-	subscriber := mqtt.NewClient(subscriberOptions)
+	if w.PubSub {
```

Comment:
I think with all those big if/else scopes, we should start to move the subscribe and publish logic into different functions. This allows us to write something like this:
```
if w.PubSub {
   startSubscribe(....)
}

Reply:
I like that approach. I'll work on it over the weekend.